### PR TITLE
feat(#154): offline mode

### PR DESCRIPTION
**Pull request by bluedrack**
Closes: #154 and #128 
Based on the work done with Yohan
I have a working setup for offline using Room.

### CODE REVIEW

**Review by lipefree — CHANGES_REQUESTED** [review]
I don't know much about Room so I can't really comment on the coding part, but it looks clean and the app is working very well. We even have some kind of persistence when going from offline to online ! Which is pretty cool

File: app/src/main/java/com/github/studydistractor/sdp/eventChat/EventChatDao.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    fun getAllMessages(eventId : String): List<Message>
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insert(messages: Message): Long
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertAll(messages: List<Message>): List<Long>
+    @Query("DELETE FROM messages")
+    fun delete()
+}
```
[the remark this answers is not in the document]
Comment (bluedrack) [reply app/src/main/java/com/github/studydistractor/sdp/eventChat/EventChatDao.kt:21]: Ok I will do it. It is clear that it is not really understable if you don't know room. Will do.

**Review by AthenaPK — COMMENTED** [review]
LGTM though it would be nice to warn the user with a toast when an action cannot be completed because we are offline. I understand though that with the limited time it may not be possible.

**Comment by bluedrack** [comment]
@AthenaPK Yes as discuss, it will be a great idea. But as you mentioned, the lack of time and coordination with you to keep the code coverage as high as possible makes this implementation quite difficult. I've tried to make it as unobtrusive as possible to the detriment of certain features, to make Adrien's and your work as easy as possible.

**Review by lipefree — APPROVED** [review]

**Pull request merged**